DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/Operations/Send/BaseObjectSenderToServer.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Speckle.Connectors.Utils.Operations;
+using Speckle.Core.Api;
+using Speckle.Core.Credentials;
+using Speckle.Core.Models;
+using Speckle.Core.Transports;
+
+namespace Speckle.Connectors.Rhino7.Operations.Send;
+
+/// <summary>
+/// Default implementation of the <see cref="IBaseObjectSender"/> which takes a <see cref="Base"/> and sends
+/// it to a server described by the parameters in the <see cref="Send"/> method
+/// </summary>
+internal sealed class BaseObjectSenderToServer : IBaseObjectSender
+{
+  private readonly Func<Account, string, ITransport> _transportFactory;
+
+  public BaseObjectSenderToServer(Func<Account, string, ITransport> transportFactory)
+  {
+    _transportFactory = transportFactory;
+  }
+
+  public async Task<string> Send(
+    Base commitObject,
+    string accountId,
+    string projectId,
+    string modelId,
+    Action<string, double?>? onOperationProgressed = null,
+    CancellationToken ct = default
+  )
+  {
+    ct.ThrowIfCancellationRequested();
+
+    onOperationProgressed?.Invoke("Uploading...", null);
+
+    // TODO: FETCHING ACCOUNTS BY ID ONLY IS UNSAFE
+    Account account =
+      AccountManager.GetAccounts().FirstOrDefault(acc => acc.id == accountId)
+      ?? throw new SpeckleAccountManagerException();
+
+    ITransport transport = _transportFactory(account, projectId);
+    var sendResult = await SendHelper.Send(commitObject, transport, true, null, ct).ConfigureAwait(true);
```

Comment:
Why are we `ConfigureAwait(true)` here? 🤔 Default is false and should be ok in this context.

Reply:
good catch. This was definitely autocomplete using the wrong bool and me not checking.